python/cudf/cudf/core/_compat.py

```diff
@@ -4,13 +4,7 @@
 from packaging import version
 
 PANDAS_VERSION = version.parse(pd.__version__)
-PANDAS_GE_133 = PANDAS_VERSION >= version.parse("1.3.3")
-PANDAS_GE_134 = PANDAS_VERSION >= version.parse("1.3.4")
-PANDAS_LT_140 = PANDAS_VERSION < version.parse("1.4.0")
-PANDAS_GE_150 = PANDAS_VERSION >= version.parse("1.5.0")
-PANDAS_LT_153 = PANDAS_VERSION < version.parse("1.5.3")
 PANDAS_EQ_200 = PANDAS_VERSION == version.parse("2.0.0")
```

Comment:
Have we determined the minimum 2.x version that we intend to support? We may choose to require 2.1.x or higher. In that case we can clean this up further, at a later time.

Reply:
@mroeschke @bdice I think we can comfortably support `2.0` as minimum version. Agree, like any release the first major release has a good number of issues that were fixed in later `2.x` versions. But imposing `2.x` itself is a breaking step change for users and I feel skipping `2.0` as the minimum version shouldn't be done at least in `24.04`. 

Coming to this PR. Can we either:

1. Hold off on this PR until we are 1-2 weeks near the burndown end for `24.04`? 

or


2. Only drop all the versions until `pre-1.5.3` and keep the version checks from `1.5.3` in `pandas_2.0_feature_branch` and then drop the remaining `1.5.3` version checks 1-2 weeks near the burndown end for `24.04`?


I feel 2 is more work, The reason I'm proposing 1 or 2 is we are yet to see how libraries and users react to `2.x` support in `24.04` dev cycle and these version checks are vital for any decision change to support both 1.5.x and 2.x simultaneously, I know that's not going to happen but I'm being cautious here.

Reply:
cc: @shwina incase you have any thoughts on the above.

Reply:
I agree, we don't want to choose a specific minor version as the minimum supported version. But I don't think we should be cautious about support 1.x/2.x simultaneously. It's likely not going to be feasible to do that.